cms/djangoapps/contentstore/views/course.py

```diff
@@ -876,6 +878,33 @@ def settings_handler(request, course_key_string):
                     courses = _remove_in_process_courses(courses, in_process_course_actions)
                 settings_context.update({'possible_pre_requisite_courses': courses})
 
+            if credit_eligibility_enabled:
+                if is_credit_course(course_key):
+                    # get and all credit eligibility requirements
+                    credit_requirements = get_credit_requirements(course_key)
+                    # pair together requirements with same 'namespace' values
+                    paired_requirements = {}
+                    for requirement in credit_requirements:
+                        requirement_data = {
+                            "name": requirement.get('name'),
+                            "display_name": requirement.get('display_name'),
+                            "criteria": requirement.get('criteria'),
+                        }
+                        paired_requirements.setdefault(requirement.get('namespace'), []).append(requirement_data)
```

Comment:
optional: Are all of these dictionary keys mandatory in a credit requirement?  If so you could shorten this even further:
```
namespace = requirement_data.pop()
paired_requirements.setdefault(namespace, []).append(requirement_data)
```